src/PhpParser/NodeVisitor/NameStmtPrefixer.php

```diff
@@ -146,6 +155,19 @@ private function prefixName(Name $name): Node
 
         $resolvedName = $this->nameResolver->resolveName($name)->getName();
 
+        // Do not prefix if there is a matching use statement.
+        $useStatement = $this->useStatements->findStatementForNode($this->namespaceStatements->findNamespaceForNode($name), $name);
+        if (
+            $useStatement !== null and
+            self::array_starts_with($resolvedName->parts, $useStatement->parts) and
+            !($parentNode instanceof ConstFetch and ($this->whitelist->isGlobalWhitelistedConstant($resolvedName->toString()) or $this->whitelist->isSymbolWhitelisted($resolvedName->toString(), true))) and
+            !($useStatement->getAttribute('parent') and $useStatement->getAttribute('parent')->alias !== null and $this->whitelist->isSymbolWhitelisted($useStatement->toString())) and
+            !($name instanceof FullyQualified) and
+            $resolvedName->parts !== ['Isolated', 'Symfony', 'Component', 'Finder', 'Finder']
+        ) {
+            return $name;
+        }
```

Comment:
There is several things I'm not too fond of here:

- the `and` (see https://github.com/kalessil/phpinspectionsea/blob/master/docs/control-flow.md#non-optimal-if-conditions)
- the long condition

the later though is hardly specific from your addition I'm already unpleased with the existing code in that regard. I'd just change a bit the indentation to make it a bit easier to follow:

```suggestion

        if (
            $useStatement !== null
            && !($name instanceof FullyQualified)
            && self::array_starts_with($resolvedName->parts, $useStatement->parts)
            && !(
                $parentNode instanceof ConstFetch
                && (
                    $this->whitelist->isGlobalWhitelistedConstant($resolvedName->toString())
                    || $this->whitelist->isSymbolWhitelisted($resolvedName->toString(), true)
                )
            )
            && !(
                $useStatement->getAttribute('parent')
                && $useStatement->getAttribute('parent')->alias !== null
                && $this->whitelist->isSymbolWhitelisted($useStatement->toString())
            )
            && $resolvedName->parts !== ['Isolated', 'Symfony', 'Component', 'Finder', 'Finder']
        ) {
            return $name;
        }
```